src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -190,18 +190,30 @@ def _get_payment_info(self, url, payload):
             "configuration": self._get_configuration(),
         }
 
-    def tokenize_card(self, order, billing_address):
+    def tokenize_card(self, order=None, billing_address=None, user=None):
         """
-        Tokenize a card for a given order
+        Tokenize a card for a given order.
+        When the parameter `user` is set when calling this method, we will
+        tokenize the card without an order and a billing address.
 
         https://docs.lyra.com/fr/rest/V4.0/api/playground/Charge/CreateToken
         """
         url = f"{self.api_url}Charge/CreateToken"
-        payload = self._get_common_payload_data(order, billing_address)
-        payload["formAction"] = "REGISTER"
-        payload["strongAuthentication"] = "CHALLENGE_REQUESTED"
-        del payload["amount"]
-        del payload["customer"]["shippingDetails"]
+        if user:
+            payload = {
+                "currency": settings.DEFAULT_CURRENCY,
+                "customer": {
+                    "email": user.email,
```

Comment:
I'll pass the user id also.

```suggestion
                    "reference": str(user.id),
                    "email": user.email,
```

Reply:
Okay I've tested with Lyra's playground and it works.